javaparser-core/src/main/java/com/github/javaparser/ast/type/ClassOrInterfaceType.java

```diff
@@ -53,8 +55,14 @@ public ClassOrInterfaceType(final ClassOrInterfaceType scope, final String name)
     }
 
     public ClassOrInterfaceType(final int beginLine, final int beginColumn, final int endLine, final int endColumn,
-            final ClassOrInterfaceType scope, final String name, final List<Type> typeArgs) {
+                                final ClassOrInterfaceType scope, final String name, final List<Type> typeArgs) {
```

Comment:
Yes, that would be a nice idea. The only issue is that it would break compatibility and this is something to consider. That said I agree it would be a nicer design.

I would say that I could merge myself a solution which maintains compatibility while for a solution which breaks compatibilityI would ask for a second opinion from another committer. I am not against that but it is just something that we should be more careful with.